application.go

```diff
@@ -233,14 +233,32 @@ func (a *Application) Run() error {
 	}()
 	a.window.SetUserPointer(unsafe.Pointer(&flutterEnginePointer))
 
+	// Init the engine
+	result := a.engine.Init(unsafe.Pointer(&flutterEnginePointer), a.config.vmArguments)
+	if result != embedder.ResultSuccess {
+		switch result {
+		case embedder.ResultInvalidLibraryVersion:
+			fmt.Printf("go-flutter: engine.Init() returned result code %d (invalid library version)\n", result)
+		case embedder.ResultInvalidArguments:
+			fmt.Printf("go-flutter: engine.Init() returned result code %d (invalid arguments)\n", result)
+		case embedder.ResultInternalInconsistency:
+			fmt.Printf("go-flutter: engine.Init() returned result code %d (internal inconsistency)\n", result)
+		default:
+			fmt.Printf("go-flutter: engine.Init() returned result code %d (unknown result code)\n", result)
+		}
+		os.Exit(1)
+	}
+
 	// Start the engine
-	result := a.engine.Run(unsafe.Pointer(&flutterEnginePointer), a.config.vmArguments)
+	result = a.engine.Run()
 	if result != embedder.ResultSuccess {
 		switch result {
 		case embedder.ResultInvalidLibraryVersion:
```

Comment:
Can this statuscode be returned by both Init AND Run?? You'd expect that if Init is fine with the lib version, then Run will be as well?

Reply:
As there is no documentation about this on the abi: https://github.com/go-flutter-desktop/go-flutter/blob/50bccfd1b0b3839c25da4c7c95095d0cebc2db32/embedder/embedder.h#L1040-L1065
I've left all cases.